src/Core/System/SalesChannel/Context/SalesChannelContextService.php

```diff
@@ -86,7 +86,9 @@ public function get(SalesChannelContextServiceParameters $parameters): SalesChan
                 $session[self::LANGUAGE_ID] = $parameters->getLanguageId();
             }
 
-            if ($parameters->getCurrencyId() !== null && !\array_key_exists(self::CURRENCY_ID, $session)) {
+            if ($parameters->getOverwriteCurrencyId() !== null) {
```

Comment:
I don't really get, how this is now a different behaviour than before :thinking: